src/pushcollector/_impl/proxy.py

```
@@ -50,11 +51,48 @@ class CollectorProxy(object):
     def __init__(self, delegate):
         self._delegate = delegate
 
+    def _translate_pushitem(self, pushitem):
+        def _get_checksum():
+            if pushitem.md5sum and pushitem.sha256sum:
+                return {"md5": pushitem.md5sum, "sha256": pushitem.sha256sum}
+            else:
+                return None
+
+        pushitem = {
+            "filename": pushitem.name,
+            "state": pushitem.state,
+            "src": pushitem.src,
+            "dest": str(pushitem.dest) if pushitem.dest else None,
+            "checksums": _get_checksum(),
+            "origin": pushitem.origin,
+            "build": pushitem.build,
+            "signing_key": pushitem.signing_key,
+        }
+
+        return pushitem
+
+    def _is_pushitem_obj(self, pushitem):
```

Could we please not do it this way at all, i.e. can we please have no direct dependency on the specific class of the given object?

I suggest that this should use duck typing.  If it's a dict then use it as a dict, if it's not a dict then go via `_translate_pushitem` and let it crash if the object is of the wrong shape.  Then it accepts any objects which are compatible with the documented API of PushItem.

An example of a concrete issue caused here is that, by putting these requirements on the object's class and module, it prevents it from working with mocks.

---

+1
I wasn't sure if we can rely on the doc or we need strict type checking.